pincer/middleware/channel_delete.py

```diff
@@ -26,11 +26,13 @@ async def channel_delete_middleware(self, payload: GatewayDispatch):
 
     channel = Channel.from_dict(construct_client_dict(self, payload.data))
 
-    if channel.guild_id in self.guilds:
-        guild = self.guilds[channel.guild_id]
-        old = filter(lambda c: c.id == channel.id, guild.channels)
-        if old:
-            guild.channels.remove(old)
+    guild = self.guilds.get(channel.guild_id)
+    if guild:
+        guild.channels = list(
+            filter(lambda c: c.id != channel.id, guild.channels)
+        )
```

Comment:
Faster and more readable imo
```suggestion
        guild.channels = [
            c for c in guild.channels if c.id != channel.id
        ]
```